examples/dreambooth/train_dreambooth.py

```diff
@@ -205,14 +206,16 @@ def parse_args(input_args=None):
     if env_local_rank != -1 and env_local_rank != args.local_rank:
         args.local_rank = env_local_rank
 
-    if args.instance_data_dir is None:
-        raise ValueError("You must specify a train data directory.")
-
     if args.with_prior_preservation:
         if args.class_data_dir is None:
             raise ValueError("You must specify a data directory for class images.")
         if args.class_prompt is None:
             raise ValueError("You must specify prompt for class images.")
+    else:
+        if args.class_data_dir is not None:
+            raise ValueError("You need not specify a data directory for class images.")
+        if args.class_prompt is not None:
+            raise ValueError("You need not specify prompt for class images.")
```

Comment:
This should be removed, it's not required when `args.with_prior_preservation` is `False`.

Reply:
This is what I added first in this PR.
This checks whether `class_data_dir` and `class_prompt` are used unintentionally nevertheless　`args.with_prior_preservation` is `False`.
( `class_data_dir` and `class_prompt` make no sense when  `args.with_prior_preservation` is `False`.)

Reply:
Aah I see. Maybe let's just do `logger.warn` here saying these won't be used if args.with_prior_preservation is False
Think raising an error is not necessary as it's not a big issue.

Reply:
I agree with your opinion.
I changed them.

Note: `warn` is deprecated.
https://github.com/python/cpython/blob/3.10/Lib/logging/__init__.py#L1491-L1493